src/pip/_internal/pep425tags.py

```diff
@@ -200,6 +200,26 @@ def is_manylinux2010_compatible():
     return pip._internal.utils.glibc.have_compatible_glibc(2, 12)
 
 
+def is_manylinux2014_compatible():
+    # type: () -> bool
+    # Only Linux and only x86-64, i686, aarch64, armv7l, ppc64, ppc64le, s390x
+    if get_platform() not in {"linux_x86_64", "linux_i686", "linux_aarch64",
+                              "linux_armv7l", "linux_ppc64", "linux_ppc64le",
```

Comment:
Comparing this list with the PEP [here](https://www.python.org/dev/peps/pep-0599/#platform-detection-for-installers), why do we have `linux_armv71` and not `linux_armhfp`?

Reply:
I forgot to mention the specifics for armhf vs armv7l, you can find discussions there:
- [warehouse](https://github.com/pypa/warehouse/pull/6684)
- [pep](https://github.com/python/peps/pull/1181)

Reply:
Yea, there's basically an update due: https://github.com/python/peps/pull/1181

Reply:
Thanks @mayeut! @pradyunsg, given that we have a little time until the next release do we wait for feedback from the PEP update before merging/approving or just do it eagerly?

Reply:
Another option would be to defer manylinux2014_armv7l after PEP approval in order not to block progress on other platforms. There's still one thing bothering me with armv7l - possible ABI conflict - as mentioned in [this warehouse comment](https://github.com/pypa/warehouse/pull/6684#issuecomment-536092525). I might comment on PEP PR as well.

Reply:
> Another option would be to defer manylinux2014_armv7l after PEP approval in order not to block progress on other platforms.

Yep, that's that @chrahunt is suggesting. :)

I'm cool with that.

Reply:
PEP has been approved with armv7l being armv7 hard-float ABI (same for warehouse).
I guess it's a bit different on `pip` ends where the hard-float ABI shall be checked before installing (i.e. declarative for PEP/warehouse, needs to be enforced in `pip`). I'll try to check what I can do to detect the ABI, not sure I can do much. What's the timeframe for next `pip`release (or when would you want this PR to land at the latest) ?

Reply:
Personally I'm OK with assuming the platform gives us the right string. Is there some use case that will not work with the current implementation?

Reply:
> Is there some use case that will not work with the current implementation?

Yes, debian `armel` will try, in certain cases, to install `armhf` wheels. A crash will happen at some point. Both are sharing the same `armv7l` machine name but ABI are incompatible.
We can see the same machine name with docker:
```
docker run --rm -it arm32v5/debian:10 /bin/bash
root@c851e727fba7:/# uname -m                  
armv7l
root@ c851e727fba7:/# apt-get update && apt-get install -y python3-distutils
root@c851e727fba7:/# python3
Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from distutils.util import get_platform
>>> get_platform()
'linux-armv7l'
```
```
docker run --rm -it arm32v7/debian:10 /bin/bash
root@e25462c8ca61:/# uname -m
armv7l
root@e25462c8ca61:/# apt-get update && apt-get install -y python3-distutils
root@e25462c8ca61:/# python3
Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from distutils.util import get_platform
>>> get_platform()
'linux-armv7l'
```

Reply:
I will add something like the following to enforce hard-float ABI:
```
import sys


def is_linux_armhf():
    try:
        with open(sys.executable, 'rb') as f:
            elf_header = f.read(40)  # read 40 first bytes of ELF header
    except OSError:
        return False
    if elf_header is None or len(elf_header) < 40:
        return False
    if isinstance(elf_header, str):
        elf_header = [ord(c) for c in elf_header]
    else:
        elf_header = list(elf_header)
    result = elf_header[0:4] == [0x7f, 0x45, 0x4c, 0x46]  # ELF magic number
    result &= elf_header[4:5] == [1]  # 32-bit ELF
    result &= elf_header[5:6] == [1]  # little-endian
    result &= elf_header[18:20] == [0x28, 0]  # ARM machine
    result &= elf_header[39:40] == [5]  # ARM EABIv5
    result &= (elf_header[37:38][0] & 4) == 4  # EF_ARM_ABI_FLOAT_HARD
    return result


print(is_linux_armhf())
```

Reply:
If we go this route it should also catch `TypeError` (in case `sys.executable` is `None`) and `IOError` (Python 2).